packages/untp-test-suite/src/core/processTestSuite.ts

```diff
@@ -68,9 +61,40 @@ export const processTestSuite = async (
   // Filter out configs with no errors
   const configContentNoError = validateCredentialResult.filter((config) => config.errors?.length === 0);
 
-  // Process each config to check data by schema
   const validatedCredentials = await Promise.all(
-    configContentNoError.map((credentialConfig) => processCheckDataBySchema(credentialConfig)),
+    configContentNoError.map(async (credentialConfig) => {
+      // Load schema
+      let schema;
+
+      if (credentialConfig.url) {
+        const { data, error, success } = await fetchData(credentialConfig.url);
+        // Handle fetch failure
+        if (!success) {
+          // Return the credential config with the error
+          return {
+            ...credentialConfig,
+            errors: [
+              {
+                instancePath: 'url',
+                message: error,
+                keyword: 'FetchError',
+                dataPath: credentialConfig.dataPath,
+              },
+            ] as ICredentialConfigError[],
+          };
+        }
+
+        schema = data;
+      } else {
+        schema = await dynamicLoadingSchemaService(credentialConfig.type, credentialConfig.version);
+      }
```

Comment:
Should the code handle loading schema inside the `dynamicLoadingSchemaService` or another? Then the core function just use, do not implement all logic in core function.

Reply:
From my understanding, should we fetch data in dynamicLoadingSchemaService() instead of processTestSuite()?